src/gui/gtk.c

```diff
@@ -3152,12 +3151,22 @@ static gboolean _scroll_wrap_scroll(GtkScrolledWindow *sw, GdkEventScroll *event
     GtkAdjustment *adj = gtk_scrolled_window_get_vadjustment(sw);
 
     const gint before = gtk_adjustment_get_value(adj);
-    gint value = before + increment*event->delta_y;
+
+    // 0.5f to actually compute everything based on the center
+    // of the row.
+    gint value = before + (0.5f + increment) * delta_y;
+
     value -= value % increment;
-    gtk_adjustment_set_value(adj, value);
-    const gint after = gtk_adjustment_get_value(adj);
 
-    if(after == before) return FALSE;
+    // adjust new value only of needed
+    if(value != before)
+      gtk_adjustment_set_value(adj, value);
+
+    // always kill propagation of scroll event otherwise Gtk will scroll
+    // the tree view and we don't want to do that. We want to let the scroll
```

Comment:
We do want to do that, because when scrolling with ctrl+alt or "mouse wheel scrolls side panel" we would otherwise get "stuck" when the mouse cursor is caught by a resizable widget (even if it doesn't have a scrollbar or enough content to use it).

But you are absolutely right that this would cause problems if the reason we don't see any change is because we haven't accumulated a full scroll_unit yet.

I would suggest changing
if(after == before) return FALSE;
to
if(event->delta_y && after == before) return FALSE; and, besides the change to using unit_deltas, leave everything else unchanged. Unless the "+ 0.5" is really needed, but I don't understand why it would help/make a difference.

The gtk_adjustment_set_value/gtk_adjustment_get_value is needed to detect if we have reached the end or start of the list/textview or if it doesn't even have a scrollbar. If that is the case, we want the scroll to be processed by the panel.

(I've entered this comment 3 times now; not sure why github doesn't list it in the PR).